### PR TITLE
other(bump): bump vllm to 0.22.0 with optimum

### PR DESCRIPTION
**Pull request by rebel-eunji**
## Major differences
1. new method - random_all_seed (device random setting)
2. torch.empty cache to no_op
3. ec_connector_output
4. changes model architecture transformers / optimum-rbln

---

### 📌 Related Issues / Tickets

* Resolves #
* Related to #

---

### ✅ Type of Change

* [ ] 🚀 Release (`release`)
* [ ] ✨ Feature (`feature`)
* [ ] 🧠 Model support (`model`)
* [ ] 🧬 Core engine changes (`core`)
* [ ] 🛠 Bug fix (`fix`)
* [ ] ⚙️ Performance improvement (`perf`)
* [ ] 🔁 Refactor or code cleanup (`refactor`)
* [ ] 📄 Documentation (`docs`)
* [x] ❓ Other (`other`): please describe

---

### 🧪 How to Test

1. Run `...`
2. Verify output: `...`
3. Edge case tested: `...`

---

### 📸 Screenshots / Logs (if applicable)

---

### 📋 Checklist

* [ ] PR title follows Conventional Commits format
* [ ] This PR is linked to an existing issue
* [ ] The test method is described, and the expected result is clearly stated
* [ ] Relevant documentation has been updated (if applicable)

---

### 💬 Notes

---

### CODE REVIEW

File: vllm_rbln/_torch_accelerator_compat.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+accelerator (NPU memory belongs to the rebel runtime), so torch 2.11 raises
+"Cannot access accelerator device when none is available" and the EngineCore
+dies during cleanup.
+
+We wrap empty_cache() to swallow that one case; other errors propagate.
+Applied via register_ops() at plugin load.
+"""
+
+# NOTE(eunji.lee):
+# required test: torch-rbln 2.11.0 + vllm 0.22.0
```
Comment (rebel-eunji) [inline vllm_rbln/_torch_accelerator_compat.py:28]: @rebel-jonghewk I'm going to test patched empty_cache with new version of torch-rbln.

File: pyproject.toml [hunk 2]
```diff
@@ -42,7 +42,8 @@ dependencies = [
     "torch",
     "torchvision",
     "torchaudio",
-    "optimum-rbln>=0.10.4",
+    "torchcodec",
+    "optimum-rbln==0.11.0a0",
```
Comment (rebel-jinhwan) [inline pyproject.toml]: nit. when do we include torch-rbln?

Comment (rebel-eunji) [reply pyproject.toml]: @rebel-jonghewk  I'm not totally sure about this part. Could you help?

File: vllm_rbln/v1/worker/utils.py [hunk 3]
```diff
@@ -288,6 +288,7 @@ def get_autobind_cpu_ids(
     Returns:
         Comma-separated string of CPU IDs, or "all" or "nobind".
     """
+    # NOTE: It should be checked.
```
Comment (rebel-yskim) [inline vllm_rbln/v1/worker/utils.py:291]: Does this comment mean that it should be checked in `dev-0.22.0`->`dev` pull request?

Comment (rebel-eunji) [reply vllm_rbln/v1/worker/utils.py:291]: No. The numa setting code is already changed in previous commit https://github.com/RBLN-SW/vllm-rbln/commit/5ea194805cfd933681b755cf3071995c265fd065 by @rebel-jinhwan.
So It would be better to check with that commit.

Comment (rebel-yskim) [reply vllm_rbln/v1/worker/utils.py:291]: hmm I think this commit is also not enough to review the whole changes... but it would be necessary if there is no performance degradation due to the NUMA consumption. did you checked it? @rebel-jinhwan

Comment (rebel-jinhwan) [reply vllm_rbln/v1/worker/utils.py:291]: @rebel-yskim Ah, I made the change because the reimplementation could be replaced in v0.22.0, and I didn't realize it was performance-critical. If you have concerns, it is fine to revert this commit in this PR. 
ps. could we add some pytest guardrails to ensure we don't hit any performance regressions by this issue?

Comment (rebel-yskim) [reply vllm_rbln/v1/worker/utils.py:291]: I'm not sure that we can check performance by pytest - because it depends on # of CPUs & processes of pytest environment, and I don't know how FSW team manages the CI/CD servers.
There are some questions that I want to know:
- Is https://github.com/RBLN-SW/vllm-rbln/commit/5ea194805cfd933681b755cf3071995c265fd065 the **"only"** changes that touches NUMA aware cpu affinity? 
- Are there any performance check CI/CDs in vllm-rbln? I think it is the better way to check guardrails, not the pytest.
- Which branch should I see if I want to check the whole changes of this feature?

Comment (rebel-jinhwan) [reply vllm_rbln/v1/worker/utils.py:291]: No, there isn't other commits than https://github.com/RBLN-SW/vllm-rbln/commit/5ea194805cfd933681b755cf3071995c265fd065 that changes NUMA settings. And, we don't have right now, but partially using guidellm for performance tracking in REBEL CI

Comment (rebel-yskim) [reply vllm_rbln/v1/worker/utils.py:291]: Then it seems that this commit looks good to me - because there is no RBLN side opt changes 👍

**Review by rebel-yskim — COMMENTED** [review]
LGTM. 
Could you tag me in EC-connector related PR in optimum-rbln?

**Comment by rebel-eunji** [comment]
@rebel-yskim 
Thanks. Here are the following PRs I may request your review on:
- Refactoring the disaggregated encoder
- optimum-rbln API to process images only

File: examples/optimum/run_qwen3_reranker.py [hunk 4]
```diff
@@ -41,17 +41,27 @@ def format_instruction(instruction, query, doc):
     return text
 
 
-def process_inputs(pairs, instruction, max_length, suffix_tokens, tokenizer):
+def process_inputs(
```
Comment (rebel-eunji) [inline examples/optimum/run_qwen3_reranker.py:44]: @rebel-jonghewk 
diffs
- force `return_dict=False` to get pure token ids, not BatchEncoding (wrapped class of huggingface) to slice list. 
- refactoring
